### DISCUSSION TITLE
On Page Load Query is vanished

### DISCUSSION THREAD

**Discussion by lalitkishorupadhyay** [TanStack/query#5380] [Q&A]
"Hi everyone, I'm using Next.js and tanstack React Query to call an API and display data on a page based on a query parameter. I'm facing an issue where, when I reload the page, the data fetched using tanstack React Query disappears from the DevTools. The data is still visible on the page, but it looks like the query is not being cached or saved properly.

I've already tried using the initialData and keepPreviousData options in useQuery, as well as using the useEffect hook to refetch the data when the component mounts or the query parameter changes, but none of these solutions have worked.

Can anyone suggest a possible cause of this issue or another solution to try? Thank you for your help!"

i have shared the screen shot of these below

after Reload
![Screenshot from 2023-05-08 15-48-13](https://user-images.githubusercontent.com/93825757/236799402-d7d4304f-561e-433d-a60a-2956b3c1a717.png)

before Reload 
![Screenshot from 2023-05-08 15-47-59](https://user-images.githubusercontent.com/93825757/236799439-44e3551e-61ce-45b4-9f25-9a3852eff85b.png)

This is my code for this page 

![code](https://user-images.githubusercontent.com/93825757/236803717-8d3bea3e-4426-4c46-8dda-81d85ca40234.png)

**Comment by TkDodo** [comment]
is your query client stable?

https://tkdodo.eu/blog/react-query-fa-qs#2-the-queryclient-is-not-stable
